### PR TITLE
fix: wrong object bindings when their addresses are duplicated with other objects

### PR DESCRIPTION
**Pull request by miyakoshi-dev**
## Description

addressing an issue where the order of trace events is reversed

## Related links

https://tier4.atlassian.net/browse/RT2-1245

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by isp-uetsuki — CHANGES_REQUESTED** [review]

File: src/caret_analyze/infra/lttng/id_remapper.py [hunk 1]
```diff
@@ -101,3 +110,28 @@ def get_object_id(
...
+                [item for item in self._addr_to_remapping_info[addr]
+                 if item.pid == pid]
+            if len(list_search) == 0:
+                return addr
+            else:
+                min_distance = abs(list_search[0].timestamp - timestamp)
+                min_distance_item = list_search[0]
+                for item in list_search:
+                    distance = abs(item.timestamp - timestamp)
+                    if min_distance is None or min_distance > distance:
```
[the remark this answers is not in the document]
Comment (miyakoshi-dev) [reply src/caret_analyze/infra/lttng/id_remapper.py]: I fixed

File: src/test/infra/lttng/test_lttng.py [hunk 2]
```diff
@@ -790,6 +1178,49 @@ def test_duplicated_events(self, mocker):
...
+            {
+                '_name': 'ros2:rcl_node_init',
+                'node_handle': HDL_NODE,
+                'rmw_handle': HDL_RMW,
+                'node_name': 'my_node_name',
+                'namespace': '/',
+                '_timestamp': 100100203,
+                '_vtid': VTID1,
+                '_vpid': VPID1
+            },
```
[the remark this answers is not in the document]
Comment (miyakoshi-dev) [reply src/test/infra/lttng/test_lttng.py]: I fixed

File: src/test/infra/lttng/test_lttng.py [hunk 3]
```diff
@@ -895,42 +1540,208 @@ def test_duplicated_events(self, mocker):
...
+            lttng.data.executors_static.df.iloc[0]['entities_collector_addr'] == HDL_ENTITIES
+
+        assert lttng.data.executors_static.df.index[1] == 2 and \
+            lttng.data.executors_static.df.iloc[1]['timestamp'] == 100101203 and \
+            lttng.data.executors_static.df.iloc[1]['entities_collector_addr'] == 1
+
+        assert lttng.data.executors_static.df.index[2] == 3 and \
+            lttng.data.executors_static.df.iloc[2]['timestamp'] == 100101205 and \
+            lttng.data.executors_static.df.iloc[2]['entities_collector_addr'] == 2
+
```
[the remark this answers is not in the document]
Comment (miyakoshi-dev) [reply src/test/infra/lttng/test_lttng.py]: I fixed

File: src/test/infra/lttng/test_lttng.py [hunk 4]
```diff
@@ -775,6 +1123,46 @@ def test_duplicated_events(self, mocker):
...
+        lttng = Lttng(events, event_filters=[], validate=False)
+
+        # lifecycle_state_machines
+        # ['node_handle']
+        assert lttng.data.lifecycle_state_machines.df.index[0] == STATE_MACHINE and \
+            lttng.data.lifecycle_state_machines.df.iloc[0]['node_handle'] == HDL_NODE
+
+        assert lttng.data.lifecycle_state_machines.df.index[1] == 1 and \
+            lttng.data.lifecycle_state_machines.df.iloc[1]['node_handle'] == 1
```
[the remark this answers is not in the document]
Comment (miyakoshi-dev) [reply src/test/infra/lttng/test_lttng.py:1136]: Cannot compare because timestamp does not exist

**Review by isp-uetsuki — APPROVED** [review]
LGTM

**Review by xygyo77 — APPROVED** [review]
image

LGTM

**Pull request merged**